internal/pkg/manifest/validate_test.go

```diff
@@ -1002,23 +1002,81 @@ func TestPipelineManifest_Validate(t *testing.T) {
 	testCases := map[string]struct {
 		Pipeline Pipeline
 
-		wantedError error
+		wantedError          error
+		wantedErrorMsgPrefix string
 	}{
 		"error if name exceeds 100 characters": {
 			Pipeline: Pipeline{
 				Name: "12345678902234567890323456789042345678905234567890623456789072345678908234567890923456789010234567890",
 			},
 			wantedError: errors.New("pipeline name '12345678902234567890323456789042345678905234567890623456789072345678908234567890923456789010234567890' must be shorter than 100 characters"),
 		},
+		"should validate pipeline stages": {
+			Pipeline: Pipeline{
+				Name: "release",
+				Stages: []PipelineStage{
+					{
+						Name: "test",
+						Deployments: map[string]*Deployment{
+							"frontend": {
+								DependsOn: []string{"backend"},
+							},
+						},
+					},
+				},
+			},
+			wantedErrorMsgPrefix: `validate "deployments" for pipeline stage test:`,
+		},
 	}
 	for name, tc := range testCases {
 		t.Run(name, func(t *testing.T) {
 			gotErr := tc.Pipeline.Validate()
 
 			if tc.wantedError != nil {
 				require.EqualError(t, gotErr, tc.wantedError.Error())
+				return
+			}
+			if tc.wantedErrorMsgPrefix != "" {
+				require.Error(t, gotErr)
+				require.Contains(t, gotErr.Error(), tc.wantedErrorMsgPrefix)
+				return
+			}
+			require.NoError(t, gotErr)
```

Comment:
nit
```suggestion
			switch {
			case tc.wantedErr != nil:
				require.EqualError(t, gotErr, tc.wantedError.Error())
			case tc.wantedErrorMsgPrefix != "":
				require.Error(t, gotErr)
				require.Contains(t, gotErr.Error(), tc.wantedErrorMsgPrefix)
			default:
				require.NoError(t, gotErr)
			}
```

Reply:
Done! I was keeping it consistent w the rest of tests but this is cleaner 😍